src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -1794,6 +1795,96 @@ document::value create_find_cursor(entity::map& map,
     return make_document();
 }
 
+document::value create_search_index(collection& coll, const document::view& operation) {
+    auto arguments = operation["arguments"];
+
+    auto raw_model = arguments["model"];
+
+    search_index_model model =
+        raw_model["name"]
+            ? search_index_model(string::to_string(raw_model["name"].get_string().value),
+                                 raw_model["definition"].get_document().value)
+            : search_index_model(raw_model["definition"].get_document().value);
+
+    return make_document(kvp("result", coll.search_indexes().create_one(model).value()));
+}
+
+document::value create_search_indexes(collection& coll, const document::view& operation) {
+    auto arguments = operation["arguments"];
+
+    auto raw_models = arguments["models"].get_array().value;
+    std::vector<search_index_model> models;
+
+    for (auto&& m : raw_models) {
+        search_index_model model =
+            m["name"] ? search_index_model(string::to_string(m["name"].get_string().value),
+                                           m["definition"].get_document().value)
+                      : search_index_model(m["definition"].get_document().value);
+        models.push_back(model);
+    }
+
+    std::vector<bsoncxx::string::view_or_value> created_indexes =
+        coll.search_indexes().create_many(models);
+
+    builder::basic::array result;
+    for (auto s : created_indexes) {
+        result.append(s);
+    }
+
+    return make_document(kvp("result", result.view()));
+}
+
+document::value drop_search_index(collection& coll, const document::view& operation) {
+    auto arguments = operation["arguments"];
+
+    auto name = string::to_string(arguments["name"].get_string().value);
+
+    coll.search_indexes().drop_one(name);
+
+    // no return value from drop_one
+    return make_document();
+}
+
+document::value list_search_indexes(collection& coll, const document::view& operation) {
+    auto arguments = operation["arguments"];
+    bsoncxx::document::view aggregation_options;
+    options::aggregate options;
+    if (arguments["aggregationOptions"]) {
+        aggregation_options = arguments["aggregationOptions"].get_document().view();
+        // search index spec tests only set batchSize, so if aggregation options exist then it must
+        // be batch size.
+        options.batch_size(aggregation_options["batchSize"].get_int32().value);
```

Comment:
```suggestion
        for (auto&& element : aggregation_options) {
            if (element.key() == "batchSize") {
                options.batch_size(element.get_int32().value);
            } else {
                throw std::logic_error{"unsupported aggregateOptions field: " +
                                       std::string(element.key())};
            }
        }
```

Suggest iterating over all elements and throwing an exception on unrecognized options. This may help inform when tests are introduced with new unrecognized options.